core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -367,7 +367,8 @@ class AclAuthorizer extends Authorizer with Logging {
     } else false
   }
 
-  @nowarn("cat=deprecation&cat=optimizer")
+  @nowarn("cat=deprecation")
+  @nowarn("cat=optimizer")
```

Comment:
The behavior of `&` was changed so that both conditions have to evaluate to true for a warning to match the exclusion.